lib/cli-engine/config-array-factory.js

```diff
@@ -623,8 +623,12 @@ class ConfigArrayFactory {
      * @private
      */
     _normalizeConfigData(configData, ctx) {
+
         validateConfigSchema(configData, ctx.name || ctx.filePath);
-        return this._normalizeObjectConfigData(configData, ctx);
+
+        const clonedConfig = JSON.parse(JSON.stringify(configData));
```

Comment:
As I mentioned in https://github.com/eslint/eslint/pull/13034#issuecomment-635098648, would you move the cloning process to the `factory._normalizeObjectConfigDataBody()` method and clone the `rules` object only? Cloning only parts that Ajv touches, so it reduces the impact of this change (we can continue to use non-JSON values in the `settings` field.).

Reply:
oh, I might have missed that comment. 

Let me check .

Reply:
I added `const clonedRulesConfig = JSON.parse(JSON.stringify((rules)));` [here](https://github.com/eslint/eslint/blob/master/lib/cli-engine/config-array-factory.js#L704)
and returning [here](https://github.com/eslint/eslint/blob/master/lib/cli-engine/config-array-factory.js#L734) as `rules: clonedRulesConfig`
It is giving the following error as

![image](https://user-images.githubusercontent.com/26347874/84588530-d3f13c00-ae45-11ea-9fbb-8161df76b344.png)


So I changed it to `const clonedRulesConfig = JSON.parse(JSON.stringify(({...rules})));`

but then it is failing the tests of [config-array-factory](https://github.com/eslint/eslint/blob/master/tests/lib/cli-engine/config-array-factory.js) like [this one](https://github.com/eslint/eslint/blob/master/tests/lib/cli-engine/config-array-factory.js#L923) and other similar ones
as this

![image](https://user-images.githubusercontent.com/26347874/84588621-aeb0fd80-ae46-11ea-9cd8-37a4e3c3091c.png)

As with this change, even if `rules` is undefined, it will create an empty object because of `{..rules}` 

![image](https://user-images.githubusercontent.com/26347874/84588940-0c464980-ae49-11ea-9d81-51622316a183.png)

 so that means those fixtures needs to changed. 

WDYT ?

Reply:
How about `rules && JSON.parse(JSON.stringify(rules))`?